model/commitqueue/commit_queue.go

```diff
@@ -30,49 +42,58 @@ func (q *CommitQueue) Enqueue(item string) error {
 	return nil
 }
 
-func (q *CommitQueue) IsEmpty() bool {
-	return len(q.Queue) == 0
-}
-
-func (q *CommitQueue) Next() string {
-	if q.IsEmpty() {
-		return ""
+func (q *CommitQueue) Next() *CommitQueueItem {
```

Comment:
The issue making sure we don't create a second test for the same item (given the commit queue Amboy job runs once a minute) is reworked here.

The method used to be to check if a patch/version existed for a given merge commit SHA. That way if we already started testing a PR we won't retest it unless the user has pushed new commits to the PR and GitHub created a new test merge commit.

The issue with this was that if something went wrong in the middle of a merge attempt (e.g. a call to the GitHub API failed) the user won't be able to trigger the commit queue again for the same PR without pushing something to it.

The current solution is to mark the commit queue as busy when an item at the head of the queue is being processed and to clear it when we are ready to process again (either because the head was dequeued or because we want to retry)